internal/controller/databasecluster_controller.go

```diff
@@ -717,7 +753,52 @@ func (r *DatabaseClusterReconciler) initWatchers(controller *builder.Builder) {
 
 			return requests
 		}),
-		builder.WithPredicates(predicate.ResourceVersionChangedPredicate{}),
+		builder.WithPredicates(predicate.ResourceVersionChangedPredicate{}, defaultPredicate),
+	)
+
+	controller.Watches(
+		&everestv1alpha1.PodSchedulingPolicy{},
+		handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, obj client.Object) []reconcile.Request {
+			attachedDatabaseClusters, err := common.DatabaseClustersThatReferenceObject(ctx, r.Client, podSchedulingPolicyNameField, "", obj.GetName())
+			if err != nil {
+				return []reconcile.Request{}
+			}
+
+			requests := make([]reconcile.Request, len(attachedDatabaseClusters.Items))
+			for i, item := range attachedDatabaseClusters.Items {
+				requests[i] = reconcile.Request{
+					NamespacedName: types.NamespacedName{
+						Name:      item.GetName(),
+						Namespace: item.GetNamespace(),
+					},
+				}
+			}
+
+			return requests
+		}),
+		// We need to filter out the events that are not in the system namespace,
+		// that is why a separate predicate.Funcs predicate is used instead of
+		// common.DefaultNamespaceFilter.
+		builder.WithPredicates(predicate.GenerationChangedPredicate{},
+			predicate.Funcs{
+				// Nothing to process on create events
+				CreateFunc: func(e event.CreateEvent) bool {
+					return false
+				},
+				// Allow update events only if the policy is in the system namespace.
+				UpdateFunc: func(e event.UpdateEvent) bool {
+					return e.ObjectOld.GetNamespace() == common.SystemNamespace
+				},
+				// Nothing to process on delete events
+				DeleteFunc: func(e event.DeleteEvent) bool {
+					return false
+				},
+				// Nothing to process on generic events
+				GenericFunc: func(e event.GenericEvent) bool {
+					return false
+				},
+			},
+		),
```

Comment:
Same predicate it used for the PSP controller, can we move this to a common function to reduce duplicate code?